dashboard/lib/service/firebase_auth.dart

```diff
@@ -71,6 +89,73 @@ class FirebaseAuthService extends ChangeNotifier {
     }
   }
 
+  Future<void> linkWithGoogle() async {
```

Comment:
nit: maybe note something like "/// Link a Google account to the signed in account (GitHub)"?